fapolicy_analyzer/ui/profiler_page.py

```diff
@@ -0,0 +1,114 @@
+# Copyright Concurrent Technologies Corporation 2021
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import glob
+import logging
+import gi
+import os
+
+from time import sleep
+from fapolicy_analyzer.ui.store import get_system_feature
+from fapolicy_analyzer.ui.ui_page import UIAction, UIPage
+from fapolicy_analyzer.ui.ui_widget import UIConnectedWidget
+from fapolicy_analyzer.ui.configs import Sizing
+from fapolicy_analyzer.ui.faprofiler import FaProfiler
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gtk
+
+
+class ProfilerPage(UIConnectedWidget, UIPage):
+    def __init__(self, data):
+        UIConnectedWidget.__init__(self, get_system_feature())
+        actions = {
+            "start": [
+                UIAction(
+                    "Start",
+                    "Start Test",
+                    "media-playback-start",
+                    {"clicked": self.on_test_activate},
+                    sensitivity_func=self.start_button_sensitivity
+                )
+            ],
+            "stop": [
+                UIAction(
+                    "Stop",
+                    "Stop Test",
+                    "media-playback-stop",
+                    {},
+                    sensitivity_func=self.stop_button_sensitivity
+                )
+            ]
+        }
+
+        UIPage.__init__(self, actions)
+        manager, window = data
+        self._fapd_profiler = FaProfiler(manager)
+        self.running = False
+
+        width = window.get_size()[0]
+        self.get_object("dirEntry").set_property("width_request", int(width * Sizing.PROFILER_TEXT_ENTRY))
+
```

Comment:
Instead of passing in the window object, it looks like you can wire up the `realize` signal. That one fires whenever the widget is added to a window. A handler like this will then do your resizing with no extra params needed.

```
    def on_profilerPage_realize(self, widget):
        window = widget.get_toplevel()
        width = window.get_size()[0]
        self.get_object("dirEntry").set_property(
            "width_request", int(width * Sizing.PROFILER_TEXT_ENTRY)
        )
```

Reply:
Also this resizing doesn't hold up if I resize my screen once the profiler page is initially rendered. You'll probably have to wire up to another signal for that one. There is probably some resized signal to listen for.

Reply:
The main issue is that the `get_toplevel` doesn't return the `Window` object from `main_window` when called from `profiler_page`, it returns a `Box` object.

Reply:
With the code above and using the `realize` signal, my `get_toplevel` returns the same object as the `window` passed into `__init__`. I think once the widget is added and the `realize` signal is fired, the toplevel points to the window, before that it points to the widgets "toplevel" which is a box because that's all it knows about.

Reply:
Fixed with my last commit